registration/src/main/proto/lobby/registration/order/events.proto

```diff
@@ -0,0 +1,127 @@
+import "lobby/registration/seat/assignment/seat_assignments.proto";
+import "google/protobuf/timestamp.proto";
+import "spine/money/money.proto";
+
+// The event propagated when an order is placed by a registrant.
+message OrderPlaced {
+    // The ID of the order.
+    OrderId order_id = 1;
+
+    // The ID of the related conference.
+    ConferenceId conference_id = 2;
+
+    // The order items representing the quantity of seats of different types at the conference.
+    repeated lobby.registration.seat.availability.SeatQuantity seat = 3;
+
+    // The expected expiration time if the reservation is not explicitly confirmed later.
+    google.protobuf.Timestamp reservation_auto_expiration = 4;
+
+    // The conference locator access code.
+    lobby.registration.ConferenceAccessCode access_code = 5;
+}
+
+// The event propagated when an order is updated.
+message OrderUpdated {
+    // The ID of the order.
+    OrderId order_id = 1;
+
+    // The order items representing the quantity of seats of different types.
+    repeated lobby.registration.seat.availability.SeatQuantity seat = 2;
+}
+
+// The event propagated when an order is partially reserved.
+message OrderPartiallyReserved {
+    // The ID of the order.
+    OrderId order_id = 1;
+
+    // The time of the reservation expiration.
+    google.protobuf.Timestamp reservation_expiration = 2;
+
+    // The order items representing the quantity of seats of different types.
+    repeated lobby.registration.seat.availability.SeatQuantity seat = 3;
+}
+
+// The event propagated when an order reservation is completed.
+message OrderReservationCompleted {
+    // The ID of the order.
+    OrderId order_id = 1;
+
+    // The time of the reservation expiration.
+    google.protobuf.Timestamp reservation_expiration = 2;
+
+    // The order items representing the quantity of seats of different types.
+    repeated lobby.registration.seat.availability.SeatQuantity seat = 3;
+}
+
+// The event propagated when an order is expired.
+message OrderExpired {
+    // The ID of the order.
+    OrderId order_id = 1;
+}
+
+// The event propagated when an order is confirmed.
+message OrderConfirmed {
+    // The ID of the order.
+    OrderId order_id = 1;
+}
+
+// The event propagated when a registrant is assigned to an order.
+message OrderRegistrantAssigned {
+    // The ID of the order.
+    OrderId order_id = 1;
+
+    // The personal info of the registrant.
+    lobby.registration.seat.assignment.PersonalInfo personal_info = 2;
+}
+
+// The event propagated when an order total price is calculated.
+message OrderTotalsCalculated {
+    // The ID of the order.
+    OrderId order_id = 1;
+
+    // The total price of the order.
+    spine.money.Money total = 2;
+
+    // The lines of the order.
+    repeated lobby.registration.order.SeatOrderLine order_line = 3;
+
+    // If the order is free of charge.
+    bool is_free_of_charge = 4;
+}
+
+// TODO:2015-12-04:alexander.litus: move to Payments BC
```

Comment:
Why cannot OrderId belong to API shared with two or more Bounded Contexts?

On Wed, 9 Dec 2015 at 11:24, AlexanderLitus <notifications@github.com>
wrote:

> In registration/src/main/proto/lobby/registration/order/events.proto
> <https://github.com/SpineEventEngine/Lobby/pull/5#discussion_r47067019>:
>
> > +// The event propagated when an order total price is calculated.
> > +message OrderTotalsCalculated {
> > +    // The ID of the order.
> > +    OrderId order_id = 1;
> > +
> > +    // The total price of the order.
> > +    spine.money.Money total = 2;
> > +
> > +    // The lines of the order.
> > +    repeated lobby.registration.order.SeatOrderLine order_line = 3;
> > +
> > +    // If the order is free of charge.
> > +    bool is_free_of_charge = 4;
> > +}
> > +
> > +// TODO:2015-12-04:alexander.litus: move to Payments BC
>
> The problem is that when I move this event to another module
> (payment-contracts, like in Contoso), there is no way to use OrderId in
> this event.
> Use string or Any instead is not very good.
> Another way is to create registration-contracts module which will contain
> OrderId, and then payment-contracts can depend on it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpineEventEngine/Lobby/pull/5/files#r47067019>.
>